tasks/terraform-cli/task.json

```
@@ -252,6 +253,27 @@
             "required": false,
             "visibleRule": "command = plan",
             "helpMarkDown": "The name to give the published plan. If provided, the plan will be published and made available on the Terraform Plan view/tab."
+        },
+        {
+            "name": "workspaceSubCommand",
+            "type": "pickList",
+            "label": "Workspace Sub Command",
+            "required": false,
+            "helpMarkDown": "The workspace sub-command to run.",
+            "options": {
+                "select": "select"
+            },
+            "properties": {
+                "EditableOptions": "False"
+            }
+        },
+        {
+            "name": "workspaceName",
+            "type": "string",
+            "label": "Workspace Name",
+            "required": false,
+            "visibleRule": "command = workspaceselect",

```

Same here, shouldn't this be `workspace`?
```suggestion
            "visibleRule": "command = workspace",

```

---

fixed. not necessarily workspace but, the workspace subcommand. `workspace list` for example doesnt take a name parameter. So the rule is based on the subcommand.